src/iop/lens.cc

```diff
@@ -1736,15 +2207,44 @@ void reload_defaults(dt_iop_module_t *module)
       lf_free(cam);
     }
   }
-#else
-  module->default_enabled = 0;
-  module->hide_enable_button = true;
 #endif
 
-  // if we have a gui -> reset corrections_done message
+  if(have_embedded_metadata)
+  {
+    // prefer embedded metadata if available
+    d->method = DT_IOP_LENS_METHOD_EMBEDDED_METADATA;
+  }
+  else if(have_lensfun)
+  {
+    d->method = DT_IOP_LENS_METHOD_LENSFUN;
+  }
+
+  if(!have_lensfun && !have_embedded_metadata)
+  {
+    module->default_enabled = 0;
+    module->hide_enable_button = true;
+  }
+  else
+  {
+    module->default_enabled = 1;
+    module->hide_enable_button = false;
+  }
+
   dt_iop_lens_gui_data_t *g = (dt_iop_lens_gui_data_t *)module->gui_data;
   if(g)
   {
+    const int menu_size = dt_bauhaus_combobox_length(g->methods_selector);
```

Comment:
Correct me if I misunderstand. Here, the combo seems to get refilled with one, two or zero options. I would suggest either hiding it, if there is only one or no options available, or setting the other option to `dt_bauhaus_combobox_entry_set_sensitive` FALSE.

What is supposed to happen when lensfun is not available and the current lens has no metadata? Could I still provide manual parameters?

Reply:
> Correct me if I misunderstand. Here, the combo seems to get refilled with one, two or zero options. I would suggest either hiding it, if there is only one or no options available

I was trying to find the better way and I ended up doing that same that is done in iop/highlights.c: https://github.com/darktable-org/darktable/blob/bb01c0e16862f710ac45b2eb57c098885118142c/src/iop/highlights.c#L2383-L2415

> I would suggest either hiding it, if there is only one or no options available

Yeah, i had also the idea to hide the combobox but I had to write in place of the combobox that we're using the embedded metadata method (right?).
Do you know if this is already done in other parts of darktable so I can do something similar (I haven't found it)?

> or setting the other option to `dt_bauhaus_combobox_entry_set_sensitive` FALSE.

I'm open to everything that from the UX perspective will be considered the better options. Thanks.


> What is supposed to happen when lensfun is not available and the current lens has no metadata?

You'll get a message in the module and all operations will be passthrough, see the initial description and:

https://github.com/darktable-org/darktable/pull/12714/commits/156e890eaafee39942496473c369e361a0eb8cdb#diff-7030639e6ad76a4c31407e865f0ca9ea360dd408d9deed9f3a7f9409e9713fcdL2730-R3246

> Could I still provide manual parameters?
If you don't have embedded metadata but lensfun is compiled you can do the same as before, nothing should have changed.

If you have darktable compiled without lensfun and embedded metadata is not available you don't have any way to play with something. The embedded metadata algorithm relies on the correction information provided by the file exif. To play with something you'll have to define all the correction data manually (and find a way to express this in the gui).

Reply:
> or setting the other option to dt_bauhaus_combobox_entry_set_sensitive FALSE.

@dterrahe I implemented (not yet pushed) this method. Looks good to me.

While testing I noticed that the entry was correctly inhibited but its color was white instead of a disabled color. There is an issue in the darktable theme and opened PR #12723. If you want to take a look at it.

Reply:
In the last push I pushed also the combobox entry disabling